server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -682,7 +683,7 @@ protected Node(final Environment initialEnvironment,
                     {
                         processRecoverySettings(settingsModule.getClusterSettings(), recoverySettings);
                         b.bind(PeerRecoverySourceService.class).toInstance(new PeerRecoverySourceService(transportService,
-                            indicesService, recoverySettings));
+                            indicesService, recoverySettings, SourceOnlyRecoveryPlannerService.INSTANCE));
```

Comment:
My plan was to change this once we get the actual recovery from snapshot merged, just to avoid any possible misbehaviour setting up `indices.recovery.use_snapshots`.